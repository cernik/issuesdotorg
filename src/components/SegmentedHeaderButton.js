// @flow
import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import {noop} from '../utils/constants';
import {Styles} from '../utils/styles';
import type {ItemStateType} from '../types';

type SegmentedHeaderButtonProps = {
  value: ItemStateType,
  selected: boolean,
  onPress: () => void,
  style: Object,
};

const SegmentedHeaderButton = ({
  value = 'all',
  selected = false,
  onPress = noop,
  style,
}: SegmentedHeaderButtonProps) => (
  <View
    style={[
      styles.button,
      {
        backgroundColor: selected ? 'gainsboro' : 'white',
      },
      style,
    ]}>
    <TouchableOpacity onPress={onPress}>
      <Text style={[Styles.control, selected ? {fontWeight: 'bold'} : {}]}>
        {value}
      </Text>
    </TouchableOpacity>
  </View>
);

const styles = StyleSheet.create({
  button: {
    borderTopWidth: 1,
    borderBottomWidth: 1,
    padding: 4,
    paddingHorizontal: 16,
    minWidth: 90,
    alignItems: 'center',
  },
});

export default SegmentedHeaderButton;
