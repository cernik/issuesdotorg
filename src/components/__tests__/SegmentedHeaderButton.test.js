// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import SegmentedHeaderButton from '../SegmentedHeaderButton';

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SegmentedHeaderButton selected value={'closed'} onPress={() => {}} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
