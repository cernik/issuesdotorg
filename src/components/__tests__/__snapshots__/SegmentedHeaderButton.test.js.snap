// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderTopWidth": 1,
        "minWidth": 90,
        "padding": 4,
        "paddingHorizontal": 16,
      },
      Object {
        "backgroundColor": "gainsboro",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "fontWeight": "500",
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      closed
    </Text>
  </View>
</View>
`;
